api/src/main/java/org/openmrs/module/idgen/IdgenUtil.java

```diff
@@ -98,4 +101,32 @@ public static long convertFromBase(String s, char[] baseCharacters) {
 		}
 		return contents;
 	}
+	
+	/**
+	 * Convenience method for getting a valid prefix from a parent location with a valid prefix attribute up the tree
+	 * 
+	 * @param location the child location. Normally the location for the current {@link UserContext}
+	 * @return prefix
+	 */
+	public static String getLocationPrefixRecursively(Location location) {
+		if (location != null) {
+			for (Object ob : location.getActiveAttributes().toArray()) {
+				LocationAttribute att = (LocationAttribute) ob;
+				// TODO : Handle hardcoding for 'Prefix', introduce a GP for it?
```

Comment:
Introduce a constant for it and create the attribute type in the module activator.

The attribute type could be named something like "Location Code".
As a description:
>"A short code that represents this location."